osu.Game.Tests/Rulesets/Scoring/ScoreProcessorTest.cs

```diff
@@ -53,5 +53,257 @@ public void TestSingleOsuHit(ScoringMode scoringMode, HitResult hitResult, int e
 
             Assert.IsTrue(Precision.AlmostEquals(expectedScore, scoreProcessor.TotalScore.Value));
         }
+
+        /// <summary>
+        /// Test to see that all <see cref="HitResult"/>s contribute to score portions in correct amounts.
+        /// </summary>
+        /// <param name="scoringMode">Scoring mode to test.</param>
+        /// <param name="hitResult">The <see cref="HitResult"/> that will be applied to selected hit objects.</param>
+        /// <param name="maxResult">The maximum <see cref="HitResult"/> achievable.</param>
+        /// <param name="expectedScore">Expected score after all objects have been judged, rounded to the nearest integer.</param>
+        /// <remarks>
+        /// This test intentionally misses the 3rd hitobject to achieve lower than 75% accuracy and exactly 50% max combo.
+        /// <para>
+        /// For standardised scoring, <paramref name="expectedScore"/> is calculated using the following formula:
+        /// 1_000_000 * ((3 * <paramref name="hitResult"/> / 4 * <paramref name="maxResult"/>) * 30% + 50% * 70%)
```

Comment:
```suggestion
        /// 1_000_000 * ((3 * <paramref name="hitResult"/> / (4 * <paramref name="maxResult"/>)) * 30% + 50% * 70%)
```

But really, I don't really like all of this maths being thrown out as an exercise for the user to calculate... What do you think about putting inline comments instead:
```
[TestCase(ScoringMode.Standardised, HitResult.Miss, HitResult.Great, 0)]
[TestCase(ScoringMode.Standardised, HitResult.Meh, HitResult.Great, 387_500)] // 50/300 * 3/4 * 300000 + 350000
[TestCase(ScoringMode.Standardised, HitResult.Ok, HitResult.Great, 425_000)] // 100/300 * 3/4 * 300000 + 350000
[TestCase(ScoringMode.Classic, HitResult.Miss, HitResult.Great, 0)]
[TestCase(ScoringMode.Classic, HitResult.Meh, HitResult.Great, 156)] // 50 * 3/4 * 4 * (1 + 1/25)
[TestCase(ScoringMode.Classic, HitResult.Ok, HitResult.Great, 312)] // 100 * 3/4 * 4 * (1 + 1/25)
```
At least that way it makes it really obvious as to what's changing and can be easily verified without digging up the hitresult values.

Reply:
> instead

Why not both?

Reply:
Sure.